dds/DCPS/transport/rtps_udp/RtpsUdpDataLink.cpp

```diff
@@ -1452,19 +1452,24 @@ RtpsUdpDataLink::send_bundled_responses(ResponseVec& responses)
 
   const RepoId no_dst = { { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 }, { { 0, 0, 0 }, 0 } };
```

Comment:
In `dds/DCPS/GuidUtils.h` there is `OpenDDS::DCPS::GUID_UNKNOWN`.

Reply:
I saw that, but how do the single {0}s expand there? Do they turn into multiple {0}s or is it just the first one?

Reply:
"If the number of initializer clauses is less than the number of members and bases (since C++17) or initializer list is completely empty, the remaining members and bases (since C++17) are initialized by their default member initializers, if provided in the class definition, and otherwise (since C++14) by empty lists, in accordance with the usual list-initialization rules (which performs value-initialization for non-class types and non-aggregate classes with default constructors, and aggregate initialization for aggregates). If a member of a reference type is one of these remaining members, the program is ill-formed."

So I think the first values get 0's and then the rest are default initialized?

Reply:
> which performs value-initialization for non-class types

which means guaranteed to be 0

Reply:
Oh, the rest are initialized by empty lists, which performs value initialization for non-class types. Sure, I can switch over to GUID_UNKNOWN.